test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetTrustTests.cs

```diff
@@ -0,0 +1,508 @@
+                // Arrange
+                var nugetConfigContent = $@"<?xml version=""1.0"" encoding=""utf-8""?>
+                    <configuration>
+                        <trustedSigners>
+                            <author name=""signer"">
+                                <certificate fingerprint=""abcdef"" hashAlgorithm=""SHA256"" allowUntrustedRoot=""false"" />
+                            </author>
+                        </trustedSigners>
+                    </configuration>";
+
+                var expectedAuthorContent = $@"Registered trusted signers:
+                     1.   signer [author]
+                          Certificate fingerprint(s):
+                            SHA256 - abcdef
+                    ";
+                File.WriteAllText(Path.Combine(pathContext.WorkingDirectory, "nuget.config"), nugetConfigContent);
+
+                //Act
+                var result = _msbuildFixture.RunDotnet(
+                    pathContext.SolutionRoot,
+                    $"nuget trust list --configfile ..{Path.DirectorySeparatorChar}nuget.config",
+                    ignoreExitCode: true);
+
+                // Assert
+                result.Success.Should().BeTrue();
+                SettingsTestUtils.RemoveWhitespace(result.Output).Should().Contain(SettingsTestUtils.RemoveWhitespace(expectedAuthorContent));
+            }
+        }
+
+        [CIOnlyTheory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task Trust_Author_RelativePathConfileFile_Success(bool allowUntrustedRoot)
+        {
+            // Arrange
+            var nugetConfigFileName = "NuGet.Config";
+            var package = new SimpleTestPackageContext();
+
+            using (SimpleTestPathContext pathContext = _msbuildFixture.CreateSimpleTestPathContext())
+            using (MemoryStream zipStream = await package.CreateAsStreamAsync())
+            using (TrustedTestCert<TestCertificate> trustedTestCert = SigningTestUtility.GenerateTrustedTestCertificate())
+            {
+                string certFingerprint = SignatureTestUtility.GetFingerprint(trustedTestCert.Source.Cert, HashAlgorithmName.SHA256);
+                string signedPackagePath = await SignedArchiveTestUtility.AuthorSignPackageAsync(trustedTestCert.Source.Cert, package, pathContext.PackageSource);
+                var config = $@"<?xml version=""1.0"" encoding=""utf-8""?>
+                    <configuration>
+                      <packageSources>
+                        <!--To inherit the global NuGet package sources remove the <clear/> line below -->
+                        <clear />
+                        <add key=""NuGetSource"" value=""{pathContext.PackageSource}"" />
+                       </packageSources>
+                      <config>
+                        <add key=""signaturevalidationmode"" value=""accept"" />
+                      </config>
+                      <trustedSigners>
+                      </trustedSigners>
```

Comment:
I assume things work fine if there's no `<trustedSigners>` node in the first place?